photutils/aperture/photometry.py

```diff
@@ -170,9 +172,20 @@ def aperture_photometry(data, apertures, error=None, mask=None,
                                    wcs=wcs)
 
     single_aperture = False
-    if isinstance(apertures, Aperture):
-        single_aperture = True
-        apertures = (apertures,)
+    try:
+        from regions import Region
+
+        from photutils.aperture.regions2aperture import regions2aperture
+        if isinstance(apertures, (Aperture, Region)):
+            single_aperture = True
+            apertures = (apertures,)
+        # convert regions to apertures
+        apertures = [regions2aperture(ap) if isinstance(ap, Region) else ap
+                     for ap in apertures]
+    except ModuleNotFoundError:
+        if isinstance(apertures, Aperture):
+            single_aperture = True
+            apertures = (apertures,)
```

Comment:
This is certainly simpler.. My thought process here was that this makes `regions` a dependency for `aperture_photometry`, even though `aperture_photometry` could work without `regions`, and `regions` is only an optional dependency of `photutils`, so a user that installs `photutils` will most likely not have `regions` installed.

Reply:
Yes, you're correct --  `regions` should not be a dependency for `aperture_photometry` unless necessary.

Since there are only two options for input (aperture or region), we can do this:

(this can go at the top of the module -- use `isort` for proper ordering):
```
from photutils.aperture import Aperture
from photutils.aperture import region_to_aperture
```

aperture_photometry:
```
    # convert regions to apertures if necessary
    apertures = [region_to_aperture(aper)
                 if not isinstance(aper, Aperture) else aper
                 for aper in apertures]

    if isinstance(apertures, Aperture):
        single_aperture = True
        apertures = (apertures,)
```

You can also use `pre-commit` for testing style issues (e.g., import order) locally before pushing:  `pre-commit run -a` 
You can also install the git commit hook with `pre-commit install`.

Reply:
Thanks for the advice, moving the `region_to_aperture` import to the top of the module certainly makes sense. I think with the approach you outlined, there's a problem when a scalar/non-list `Aperture` is passed, since the `for aper in apertures` will fail. Moving the `if isinstance(apertures, Aperture): ...` block before the conversion loop also doesn't work since it doesn't catch single `Region` objects (if that's something we want to support).

I'm not really sure how to resolve this without a try/catch block of some kind, either for `StopIteration` or `ModuleNotFoundError`. If you have more thoughts I would be glad to hear them!

Thanks for the note on `pre-commit`. I do have it installed, but it hasn't been catching line length issues, and I also have an aggressive formatter that runs on save in my editor, which I forgot to turn off at one point. It shouldn't be a problem any more!

Reply:
Ah, yes. We could do something like this before the list comprehension converter:

```
    single_aperture = False
    if not isinstance(apertures, list):
        single_aperture = True
        apertures = (apertures,)
```

or even:
```
apertures = np.atleast_1d(apertures)
single_aperture = True if len(apertures) == 1 else False
```